tools/hilow.wdl

```diff
@@ -0,0 +1,295 @@
+        prefix: "Prefix for output regions file. The extension `.bed` will be added."
+    }
+
+    input {
+        File annotation
+        String prefix = "gene_regions"
+    }
+
+    String base = basename(annotation, ".gz")
+
+    #@ except: LineWidth
+    command <<<
+        set -euo pipefail
+
+        gunzip -c ~{annotation} > ~{base} \
+           || ln -sf ~{annotation} ~{base}
+
+        sed -i 's/ /\t/g' ~{base}
+
+        awk -F\\t '{ if($3 == "transcript" && $1 !~ "chrM") 
+            print $1 "++" $4 "++" $5 "++" substr($12,2,length($12)-5) "|" substr($10,2,length($10)-3)}' \
+            ~{base} \
+            | grep -v "unassigned" \
+            | sort \
+            | uniq \
+            | sed s/\+\+/\\t/g \
+            | sort -k1,1 -k2,2n > ~{prefix}.bed
+    >>>
+
+    output {
+        File genes = prefix + ".bed"
+    }
+
+    runtime {
+        cpu: 1
+        memory: "4 GB"
+        maxRetries: 1
+        container: "ubuntu:22.04"
+    }
+}
+
+task qcreport {
+    meta {
+        description: "Write QC report for Hi-C experiment"
+        outputs: {
+            qc_report: "QC report file"
+        }
+    }
+
+    parameter_meta {
+        all_valid_pairs_stats: "HiC-Pro allValidPairs mergestat statistics file"
+        mapping_stats_read1: "HiC-Pro mapping statistics for read 1"
+        mapping_stats_read2: "HiC-Pro mapping statistics for read 2"
+        pairing_stats: "HiC-Pro pairing statistics"
+        peaks_bed: "Peaks in BED format"
+        fithichip_bed: "FithiChIP interactions in BED format"
+        fithichip_q01_bed: "FithiChIP interactions with q-value < 0.01 in BED format"
+        prefix: "Prefix for the output file"
+    }
+
+    input {
+        File all_valid_pairs_stats
+        File mapping_stats_read1
+        File mapping_stats_read2
+        File pairing_stats
+        File? peaks_bed
+        File? fithichip_bed
+        File? fithichip_q01_bed
+        String prefix = basename(all_valid_pairs_stats, "_allValidPairs.mergestat")
+    }
+
+    Int disk_size_gb = ceil(
+        size(all_valid_pairs_stats, "GiB") +
+        size(mapping_stats_read1, "GiB") +
+        size(mapping_stats_read2, "GiB") +
+        size(pairing_stats, "GiB") +
+        size(peaks_bed, "GiB") +
+        size(fithichip_bed, "GiB") +
+        size(fithichip_q01_bed, "GiB")
+    ) + 2
+
+    command <<<
+        python /usr/local/bin/qc_hic.py \
+            --all_valid_pairs_stats ~{all_valid_pairs_stats} \
+            --mapping_stats_read1 ~{mapping_stats_read1} \
+            --mapping_stats_read2 ~{mapping_stats_read2} \
+            --pairing_stats ~{pairing_stats} \
+            ~{if defined(peaks_bed) then "--peaks_bed ~{peaks_bed}" else ""} \
+            ~{if defined(fithichip_bed) then "--fithichip_bed ~{fithichip_bed}" else ""} \
+            ~{(
+                if defined(fithichip_q01_bed)
+                then "--fithichip_q01_bed ~{fithichip_q01_bed}"
+                else ""
+            )} \
```

Comment:
Technically, it is possible and equivalent to drop the `if` wrapping around these interpolations. I think we've discussed this "feature" of WDL before with some mixed feelings... The verbosity here is probably clearer, but it is quite verbose...